woocommerce/woocommerce-android
# Convert markReviewAsRead into suspendable function

Merge Instructions:
1. Review [the PR in FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2195)
2. Review this PR
3. Merge FluxC
4. Update FluxC hash
5. Remove "Do not merge" label
5. Merge this PR

This PR refactors markReviewAsRead into a suspendable function - removes usage of the event bus.

There are two reasons for this change

1. [The event bus architecture is considered as legacy/deprecated](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture)
2. WCAndroid app uses stateful repositories which need to be registered to the event bus - this leads to memory leaks when the repository doesn't unregister from the event bus. It also causes side-effects when multiple instances of a repository are created - eg. duplicate tracking events.

To Test:
1. Open Reviews List
2. (If you don't have any unread reviews, publish a new review from the web)
3. Tap on an unread review
4. Go back
5. Notice the review is marked as read
6. (If you don't have any unread reviews, publish a new review from the web)
7. Tap on the ☑️ icon in the top right corner
8. Notice all unread reviews get marked as read
